ipa-core/src/sharding.rs

```diff
@@ -1,9 +1,28 @@
-use std::fmt::{Display, Formatter};
+use std::{
+    fmt::{Debug, Display, Formatter},
+    num::TryFromIntError,
+};
 
 /// A unique zero-based index of the helper shard.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ShardIndex(u32);
 
+#[derive(Debug, Copy, Clone)]
+pub struct Shard {
```

Comment:
I might call this `Sharded`? I like how `SomeType<Sharded>` reads better than `SomeType<Shard>`. Right now it's purely shard identity information, but `Context<Sharded>` feels like a better home for a get-channel-to-shard method than `Context<Shard>`.

(`NoSharding` could be renamed to `NotSharded` for symmetry, but that's a super nit.)

Reply:
I don't mind renaming it, We already have `Sharded` struct inside test infrastructure, but it may not be a big deal to have two

Reply:
Too annoying to have two, renamed `Sharded` struct defined inside `TestWorld` to `WithShards`